File: packages/ui/src/schemas/health.ts

import { z } from 'zod';

export const ReadinessSchema = z.object({
    status: z.enum(['ready', 'degraded', 'not_ready']),
    service: z.string(),
    timestamp: z.string(),
    dependencies: z.record(z.string()),
    message: z.string().nullable().optional(),
});

export type Readiness = z.infer<typeof ReadinessSchema>;

File: packages/api/src/schemas/health.py
"""Health check response schemas per REQ-5-006."""

from pydantic import BaseModel


class LivenessResponse(BaseModel):
    """Response for /health/live -- process-level liveness only."""

    status: str
    service: str
    timestamp: str


class ReadinessResponse(BaseModel):
    """Response for /health/ready -- includes dependency checks.

    ``status`` is one of: ``ready``, ``degraded``, ``not_ready``.
    """

    status: str
    service: str
    timestamp: str
    dependencies: dict[str, str]
    message: str | None = None
